kubernetes-client/src/Kubernetes/Client/Auth/Exec.hs

```diff
@@ -0,0 +1,95 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module Kubernetes.Client.Auth.Exec
+  ( Exec
+  , execAuth
+  , decodeGkeGcloudAuthPluginToken
+  ) where
+
+import Data.Aeson (FromJSON (..), eitherDecodeStrict)
+import Data.Maybe (fromMaybe)
+import Data.Text (Text, unpack)
+import Data.Typeable (typeOf)
+import GHC.Generics (Generic)
+import Kubernetes.Client.Auth.Internal.Types (DetectAuth)
+import Kubernetes.Client.KubeConfig (AuthInfo (..), ExecConfig (..), ExecEnvVar (..))
+import Kubernetes.OpenAPI.Core
+import qualified Data.ByteString.Char8 as ByteString
+import qualified System.Process as Process
+
+-------------------------------------------------------------------------------
+-- Authentication using Auth
+
+-- | The command that is to be run to obtain the token
+data Exec = Exec
+  { config :: ExecConfig
+  , decodeToken :: String -> Either String Text
+  }
+
+runExec :: Exec -> IO Text
+runExec Exec { config = ExecConfig {..}, decodeToken } = do
+  let
+    environment =
+      fmap (fmap (\ExecEnvVar {..} -> (unpack name, unpack value))) env
+
+    process =
+      (Process.proc (unpack command) (fmap unpack $ fromMaybe [] args))
+        { Process.env = environment }
+
+  output <- Process.readCreateProcess process ""
```

Comment:
We'll have to see if upstream has any opinions about how best to run commands (I don't have any problems with this).